modules/control/controller/lat_controller.cc

```diff
@@ -610,18 +611,37 @@ void LatController::ComputeLateralErrors(
   debug->set_heading_error(heading_error);
 
   auto lateral_error_dot = linear_v * std::sin(heading_error);
+  auto lateral_error_dot_dot = linear_a * std::sin(heading_error);
 
   if (FLAGS_reverse_heading_control) {
     if (VehicleStateProvider::Instance()->gear() ==
         canbus::Chassis::GEAR_REVERSE) {
       lateral_error_dot = -lateral_error_dot;
+      lateral_error_dot_dot = -lateral_error_dot_dot;
     }
   }
   debug->set_lateral_error_rate(lateral_error_dot);
 
-  auto heading_error_rate =
-      angular_v - target_point.path_point().kappa() * target_point.v();
-  debug->set_heading_error_rate(heading_error_rate);
+  debug->set_lateral_acceleration(lateral_error_dot_dot);
+
+  debug->set_lateral_jerk(
+      (debug->lateral_acceleration() - previous_lateral_acceleration_) / ts_);
+  previous_lateral_acceleration_ = lateral_error_dot_dot;
+
+  debug->set_heading_rate(angular_v);
+  debug->set_ref_heading_rate(target_point.path_point().kappa() *
+                              target_point.v());
+
+  debug->set_heading_error_rate(debug->heading_rate() -
+                                debug->ref_heading_rate());
+
+  debug->set_heading_acceleration(
```

Comment:
1. Not sure if calling "heading jerk" is a good idea, but I see the point of calling it here, so please add detailed calculation in notes and also saying A.K.A "yaw jerk". 

2. Since we add both "ref heading rate", "reference heading rate" as well as "heading error rate", why not add all them for both acceleration and jerk level? So we and have a total profiling for both the ones from planning and from the real vehicle feedback?

Reply:
I see the comments been added. Thanks.